index.html

```diff
@@ -1,12 +1,57 @@
 <!DOCTYPE html>
-<html lang="en">
+<html>
 <head>
-    <meta charset="UTF-8">
-    <meta http-equiv="X-UA-Compatible" content="IE=edge">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>Weather Report</title>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width">
+    <title>Connecting HTML, CSS, and JS Demo</title>
+    <link href="styles/index.css" rel="stylesheet" type="text/css" />
+    <script src="https://kit.fontawesome.com/43bce04ddf.js" crossorigin="anonymous"></script>
 </head>
 <body>
+    <div class="container">
```

Comment:
Your html is well organized, but makes use of many `div`s. To make it more semantic, consider sectioning elements such as `header` and `section`, for instance.